os/generic_unix_base.h

```diff
@@ -10,11 +10,15 @@
  *
  */
 
-#ifndef __OS_LINUX_BASE__
-#define __OS_LINUX_BASE__
+#ifndef __OS_GENERIC_UNIX_BASE__
+#define __OS_GENERIC_UNIX_BASE__
 
 #if __has_include(<sys/sysmacros.h>)
 #include <sys/sysmacros.h>
+#if defined(__FreeBSD)
```

Comment:
shouldn't this be `__FreeBSD__` ?

Reply:
also it's weird it's inside the `__has_include(<sys/sysmacros.h>)`, I'd move it outside.